src/main/java/net/neoforged/neoforge/event/entity/player/PlayerEnchantItemEvent.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.event.entity.player;
+
+import java.util.List;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.enchantment.EnchantmentInstance;
+
+/**
+ * This event captures when a player action enchants an item.
```

Comment:
This sentence (a) doesn't make sense and (b) isn't descriptive enough.  Suggested:
```
This event fires when a player enchants an item in an enchanting table, after {@link IItemExtension#applyEnchantments} has been called.
<p>
This event is only fired on the logical server.
```

Reply:
I would like to make a counterpoint. An event class is not the same as the NeoForge hooks.  Any modder can fire any event whenever they feel it appropriate. Therefore, the event javadoc should convey to the modder what it does and how other modders expect to see it. In contrast, the NF hooks are embedded in the vanilla code and do have a specific relation to vanilla behavior and therefore, it does make sense to have your proposed detail there. 

I know not all event javadocs adhere to that paradigm, but I think they should. The distinction between the event and their usage should be that, distinct. 

As a practical example, imagine I create a feature where throwing lapis and an item into a cauldron gives it a random enchant. I would fire this event to capture the enchant action, but the javadoc would incorrectly describe what just happened and listeners might assume the user is in a GUI because of it.

Reply:
Then you would still make this change, but simply omit `in an enchanting table`.